kin/client.py

```diff
@@ -183,6 +183,59 @@ def get_transaction_data(self, tx_hash, simple=True):
             return SimplifiedTransaction(raw_tx, self.kin_asset)
         return raw_tx
 
+    def get_account_tx_history(self, address, amount=10, descending=True, cursor=None, simple=True):
+        """
+        Get the transaction history for a given account.
+        :param str address: The public address of the account to query
+        :param int amount: The maximum number of transactions to get
+        :param bool descending: The order of the transactions, True will start from the latest one
+        :param int cursor: The horizon paging token
+        :param bool simple: Should the returned txs be simplified, if True, complicated txs will be ignored
+        :return: A list of transactions
+        :rtype: list
+        """
+
+        if not is_valid_address(address):
+            raise ValueError('invalid address: {}'.format(address))
+
+        if amount <= 0:
+            raise ValueError('Limit must be bigger than 0')
+
+        tx_list = []
+
+        requested_amount = amount if amount < MAX_RECORDS_PER_REQUEST else MAX_RECORDS_PER_REQUEST
+        params = {
+            'limit': requested_amount,
+            'order': 'desc' if descending else 'asc'
+        }
+
+        # cursor is optional
+        if cursor is not None:
+            params['cursor'] = cursor
+
+        horizon_response = self.horizon.account_transactions(address, params)
+
+        for transaction in horizon_response['_embedded']['records']:
+            raw_tx = RawTransaction(transaction)
+            if simple:
+                try:
+                    simple_tx = SimplifiedTransaction(raw_tx, self.kin_asset)
+                    tx_list.append(simple_tx)
+                except KinErrors.CantSimplifyError:
+                    pass
+            else:
+                tx_list.append(raw_tx)
+            last_cursor = transaction['paging_token']
+
+        remaining_txs = amount - len(tx_list)
+        # if we got all the txs that we wanted, or there are no more txs
+        if remaining_txs <= 0 or len(horizon_response['_embedded']['records']) < amount:
+            return tx_list
+        # If there are anymore transactions, recursively get the next transaction page
+        return tx_list.extend(self.get_account_tx_history(address, remaining_txs, descending, last_cursor, simple))
```

Comment:
Please notice that recursion is an expensive operation, if you think that this method can be called many times, it is better to change the method.